sonar-python-plugin/src/main/java/org/sonar/plugins/python/PythonScanner.java

```diff
@@ -104,17 +104,22 @@ private void scanFile(InputFile inputFile) {
         .message(e.getMessage())
         .save();
     }
-    PyFile pyFile = new org.sonar.python.frontend.PythonParser().parse(fileContent);
 
     for (PythonCheck check : checks.all()) {
       check.scanFile(visitorContext);
     }
-    SubscriptionVisitor.analyze(checks.all(), visitorContext, pyFile);
-    saveIssues(inputFile, visitorContext.getIssues());
 
-    PythonHighlighter pythonHighlighter = new PythonHighlighter(context, inputFile);
-    pyFile.accept(pythonHighlighter);
-    pythonHighlighter.getNewHighlighting().save();
+    try {
+      PyFile pyFile = new org.sonar.python.frontend.PythonParser().parse(fileContent);
+      SubscriptionVisitor.analyze(checks.all(), visitorContext, pyFile);
+      PythonHighlighter pythonHighlighter = new PythonHighlighter(context, inputFile);
+      pyFile.accept(pythonHighlighter);
+      pythonHighlighter.getNewHighlighting().save();
+    } catch (RecognitionException e) {
+      // Ignore, we should not raise a parsing error twice (once for each parser)
```

Comment:
What do you think about adding `DEBUG` log message?

Reply:
I had to rebase against the master. Now, we need both parsers to compute metrics. I had to [restructure the code](https://github.com/SonarSource/sonar-python/pull/252/commits/55903b135010295800d454f470d0766340f7e32a#diff-3939b2c0f41ee15dc69c575494495142).